runtime/init-container/src/init.c

```diff
@@ -756,6 +763,21 @@ static uint32_t spawn_new_process(struct new_process_args* new_proc_args,
                     ret = errno;
                     goto out_err;
                 }
+                proc_desc->redirs[fd].buffer.closed = false;
+
+                if (add_epoll_fd_desc(&proc_desc->redirs[fd],
```

Comment:
I moved epoll subscription above creating the child process to properly get EPOLLHUP notification if the process redirects its output. For example running `/bin/sh -c 'date > date.txt'`.